File: 20171204-java-banking-kata/src/main/java/com/katas/Transaction.java
package com.katas;

import java.time.LocalDate;

public class Transaction {

    private LocalDate date;
    private Integer amount;

    Transaction(LocalDate date, Integer amount) {
        this.date = date;
        this.amount = amount;
    }

    public LocalDate getDate() {
        return date;
    }

    public Integer getAmount() {
        return amount;
    }
}

File: 20171201-kotlin-banking-kata/src/main/kotlin/com/katas/Transaction.kt
package com.katas

import java.time.LocalDate

sealed class Transaction(val date: LocalDate, val amount: Int, val balance: Int) {
    class Deposit(date: LocalDate, amount: Int, balance: Int) : Transaction(date, amount, balance)
    class Withdraw(date: LocalDate, amount: Int, balance: Int) : Transaction(date, amount, balance)
}